runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -372,9 +325,12 @@ SerializedAOTMethod::SerializedAOTMethod() :
    }
 
 bool
-SerializedAOTMethod::isValid() const
+SerializedAOTMethod::isValid(const JITServerAOTCacheReadContext &context) const
    {
-   return _optLevel < TR_Hotness::numHotnessLevels;
+   return _optLevel < TR_Hotness::numHotnessLevels &&
+          (definingClassChainId() < context._classChainRecords.size()) &&
+          (aotHeaderId() < context._aotHeaderRecords.size()) &&
```

Comment:
Why the asymmetry here?  We don't check that  `context._aotHeaderRecords.size[aotHeaderId()]` is non zero